tests/storage/concurrent_vm_boot/conftest.py

```
@@ -0,0 +1,95 @@
+
+        yield vms
+    finally:
+        errors = run_vms_parallel(
+            vms=vms,
+            func=lambda vm: vm.clean_up(),
+            label="Failed to clean up VM",
+        )
+        # Force GC to reclaim thread-local state and deferred object cleanup after concurrent workload.
+        gc.collect()
+        if errors:
+            LOGGER.error(f"Failed to clean up VMs: {errors}")
```

Ruff (0.16.1)

[warning] Logging statement uses f-string

(G004)